client/src/components/Protected.tsx

```diff
@@ -1,14 +1,21 @@
 import { useEffect } from "react";
 import { useNavigate } from "react-router-dom";
 import { useLoginContext } from "../contexts/loginContext";
+import { checkStatus } from "@/utils/authService";
 
-const withAuth = (WrappedComponent: React.FC) => {
+const withAuth = (WrappedComponent: React.FC, canRedirect: boolean) => {
   const ProtectedComponent = (props: object) => {
-    const { loginStatus } = useLoginContext();
+    const { loginStatus, setLoginStatus } = useLoginContext();
     const navigate = useNavigate();
     useEffect(() => {
       if (!loginStatus) {
-        navigate("/login");
+        if (checkStatus()) {
+          setLoginStatus(true);
+        } else {
+          if (canRedirect) {
+            navigate("/login");
```

Comment:
shall we handle the redirection on the server?
https://expressjs.com/en/5x/api.html#res.redirect